### PR TITLE
task/HW-55479-US12-Add-Transfer-Method-Send-event-when-general-error-occurs

### PR DESCRIPTION
**Pull request by asenthil-altmtrk**

### CODE REVIEW

File: Common/Tests/Insights/HyperwalletInsightsTests.swift [hunk 1]
```diff
@@ -62,6 +62,19 @@ class HyperwalletInsightsTests: XCTestCase {
         XCTAssertNotNil(mockInsights.params[currencyTag], "Params should have currency")
     }
 
+    func testtrackError() {
```
Comment (jsaini-hw) [inline Common/Tests/Insights/HyperwalletInsightsTests.swift]: testTrackError

File: Common/Tests/Insights/HyperwalletInsightsTests.swift [hunk 2]
```diff
@@ -77,6 +90,19 @@ class HyperwalletInsightsTests: XCTestCase {
         HyperwalletInsights.shared.trackImpression(pageName: pageName, pageGroup: pageGroup, params: params)
         XCTAssertNotNil(hyperwalletInsights?.insights, "Insights should be reloaded if nil")
     }
+
+    func testtrackError_InsightsNotInitialized() {
```
Comment (jsaini-hw) [inline Common/Tests/Insights/HyperwalletInsightsTests.swift]: testTrackError_InsightsNotInitialized()

Comment (asenthil-altmtrk) [reply Common/Tests/Insights/HyperwalletInsightsTests.swift]: Changed to Uppercase

File: Common/Tests/Insights/HyperwalletInsightsTests.swift (hunk 1 above)
Comment (asenthil-altmtrk) [reply Common/Tests/Insights/HyperwalletInsightsTests.swift]: Changed to Uppercase

File: Common/Sources/View/Error/ErrorView.swift [hunk 3]
```diff
@@ -76,10 +76,20 @@ public final class ErrorView {
     }
 
     private func unexpectedError() {
+      if let pageName = self.pageName, let pageGroup = self.pageGroup {
+       let errorInfo = ErrorInfoBuilder(type: self.errorTypeException,
+                                        message: self.error.errorDescription ?? "")
```
Comment (ddhar-hw) [inline Common/Sources/View/Error/ErrorView.swift]: message should be `error.getHyperwalletErrors()?.errorList?.first?.message`

File: Common/Sources/View/Error/ErrorView.swift [hunk 4]
```diff
@@ -76,10 +76,20 @@ public final class ErrorView {
...
+                .fieldName(fieldName: "")
```
Comment (ddhar-hw) [inline Common/Sources/View/Error/ErrorView.swift]: Missing `code` - should be `code: error.getHyperwalletErrors()?.errorList?.first?.code`

File: Common/Sources/View/Error/ErrorView.swift [hunk 5]
```diff
@@ -79,10 +79,21 @@ public final class ErrorView {
     }
 
     private func unexpectedError() {
+      if !pageName.isEmpty, !pageGroup.isEmpty {
```
Comment (jsaini-hw) [inline Common/Sources/View/Error/ErrorView.swift]: why check for empty? no need of this

File: Common/Sources/View/Error/ErrorView.swift [hunk 6]
```diff
@@ -79,10 +79,21 @@ public final class ErrorView {
...
+       let errorInfo = ErrorInfoBuilder(type: self.errorTypeException,
+                                        message: error.getHyperwalletErrors()?.errorList?.first?.message ?? "")
+                .fieldName(fieldName: "")
```
Comment (jsaini-hw) [inline Common/Sources/View/Error/ErrorView.swift]: No need to add empty fieldName

File: Transfer/Sources/CreateTransferController.swift [hunk 7]
```diff
@@ -42,7 +42,6 @@ final class CreateTransferController: UITableViewController {
         (TransferButtonCell.self, TransferButtonCell.reuseIdentifier),
         (TransferNotesCell.self, TransferNotesCell.reuseIdentifier)
     ]
-
```
Comment (jsaini-hw) [inline Transfer/Sources/CreateTransferController.swift]: why this change?

File: TransferMethod/Sources/AddTransferMethodController.swift [hunk 8]
```diff
@@ -48,6 +48,8 @@ final class AddTransferMethodController: UITableViewController {
     private var presenter: AddTransferMethodPresenter!
     private var widgets = [AbstractWidget]()
     private var isCalledByScrollToRow = false
+    private let pageName = "transfer-method:add:collect-transfer-method-information"
+    private let pageGroup = "transfer-method"
```
Comment (jsaini-hw) [inline TransferMethod/Sources/AddTransferMethodController.swift]: Why do you need them here? These are already defined in Presenter

File: TransferMethod/Sources/ListTransferMethodController.swift [hunk 9]
```diff
@@ -34,7 +34,6 @@ final class ListTransferMethodController: UITableViewController {
                                                                               .localized())
     private lazy var addAccountButton: UIButton = view.setUpEmptyListButton(text: "add_account_title".localized(),
                                                                             firstItem: emptyListLabel)
-
```
Comment (jsaini-hw) [inline TransferMethod/Sources/ListTransferMethodController.swift]: Why this change?

File: Common/Sources/View/Error/ErrorView.swift (hunk 3 above)
Comment (asenthil-altmtrk) [reply Common/Sources/View/Error/ErrorView.swift]: Message was changed

File: Common/Sources/View/Error/ErrorView.swift (hunk 4 above)
Comment (asenthil-altmtrk) [reply Common/Sources/View/Error/ErrorView.swift]: Code modified

File: Common/Sources/View/Error/ErrorView.swift (hunk 5 above)
Comment (asenthil-altmtrk) [reply Common/Sources/View/Error/ErrorView.swift]: Removed

File: Common/Sources/View/Error/ErrorView.swift (hunk 6 above)
Comment (asenthil-altmtrk) [reply Common/Sources/View/Error/ErrorView.swift]: Removed fieldname

File: Transfer/Sources/CreateTransferController.swift (hunk 7 above)
Comment (asenthil-altmtrk) [reply Transfer/Sources/CreateTransferController.swift]: Empty space added. This happens when i added and removed the constant(Pagename , Pagegroup)

File: TransferMethod/Sources/AddTransferMethodController.swift (hunk 8 above)
Comment (asenthil-altmtrk) [reply TransferMethod/Sources/AddTransferMethodController.swift]: Removed -  This was added before showError method in presenter  was modified to pass pagename and pagegroup to controller

File: TransferMethod/Sources/ListTransferMethodController.swift (hunk 9 above)
Comment (asenthil-altmtrk) [reply TransferMethod/Sources/ListTransferMethodController.swift]: Added the empty space. This happens when i added and removed the constant(Pagename , Pagegroup)
